be/src/vec/functions/function_bitmap.cpp

```diff
@@ -887,6 +942,40 @@ struct BitmapToString {
     }
 };
 
+struct NameBitmapToBase64 {
+    static constexpr auto name = "bitmap_to_base64";
+};
+
+struct BitmapToBase64 {
+    using ReturnType = DataTypeString;
+    static constexpr auto TYPE_INDEX = TypeIndex::BitMap;
+    using Type = DataTypeBitMap::FieldType;
+    using ReturnColumnType = ColumnString;
+    using Chars = ColumnString::Chars;
+    using Offsets = ColumnString::Offsets;
+
+    static Status vector(const std::vector<BitmapValue>& data, Chars& chars, Offsets& offsets) {
+        size_t size = data.size();
+        offsets.resize(size);
+        chars.reserve(size);
+        for (size_t i = 0; i < size; ++i) {
+            BitmapValue& bitmap_val = const_cast<BitmapValue&>(data[i]);
+            auto ser_size = bitmap_val.getSizeInBytes();
+            char ser_buff[ser_size];
+            bitmap_val.write_to(ser_buff);
+
+            int encoded_size = (int)(4.0 * ceil((double)ser_size / 3.0));
+            char dst[encoded_size];
+            memset(dst, 0, encoded_size);
+            int outlen =
+                    base64_encode((const unsigned char*)ser_buff, ser_size, (unsigned char*)dst);
+            DCHECK(outlen > 0);
+            StringOP::push_value_string(std::string_view(dst, outlen), i, chars, offsets);
```

Comment:
IF the output string will copy the column, then we could reuse the buffer. And realloc a new buffer if it size is less than needed.